### DISCUSSION TITLE
Building a client: signing in with MFA returning InvalidToken

### DISCUSSION THREAD

**Discussion by alexispurslane** [stoatchat/discussions#607] [Q&A] [answered]
I'm correctly completing the first stage of the login, sending the email and password and getting back the multi-factor auth ticket and allowed authentication methods. Then I'm sending back what I'm pretty sure is the correct response:

```json
{
    "mfa_ticket":"<ticket pulled from response.ticket>",
    "friendly_name":"<some name>",
    "mfa_respose":{"totp_code":"<seven digit code>"}
}
````

but the server gives an Unauthorized, InvalidToken response no matter what I do. I'm pulling the mfa_ticket directly from the ticket provided in response to the user's email and password, so it should be correct, is there some further processing I have to do?

**Comment by Rexogamer** [comment, marked as answer]
Not sure if this is the issue with your code or just a typo in your example, but the third one should be `mfa_response`

**Reply by alexispurslane** [reply]
oh fuck that'd be it. perennial can't speller over here :smiling_face_with_tear: thanks!
